### ISSUE TITLE
Elasticsearch: bad transfer encoding in HTTP response

### ISSUE THREAD

**Issue by AnvilStriker** [atlassian/localstack#123]
(Copied from https://stackoverflow.com/questions/44421921/curl-localstack-elasticsearch-bad-transfer-encoding-in-http-response)

I'm trying to use curl to access Elasticsearch in the atlassianlabs/localstack:0.4.3 Docker container, but no matter which of its URLs I hit, the response is always rejected by curl:
```
% curl http://localhost:4571/
curl: (56) Illegal or missing hexadecimal sequence in chunked-encoding
```
If I use verbose mode, show headers and raw output, I see this:
```
% curl -iv --raw http://localhost:4571/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 4571 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:4571
> User-Agent: curl/7.51.0
> Accept: */*
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
HTTP/1.0 200 OK
< Server: BaseHTTP/0.3 Python/2.7.13
Server: BaseHTTP/0.3 Python/2.7.13
< Date: Wed, 07 Jun 2017 19:42:41 GMT
Date: Wed, 07 Jun 2017 19:42:41 GMT
< content-type: application/json; charset=UTF-8
content-type: application/json; charset=UTF-8
< content-encoding: gzip
content-encoding: gzip
< transfer-encoding: chunked
transfer-encoding: chunked
< Content-Length: 327
Content-Length: 327

< 
{
  "name" : "VEU9JNc",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "XC-fLqGVRiOKIUFaXJK5ng",
  "version" : {
    "number" : "5.3.0",
    "build_hash" : "3adb13b",
    "build_date" : "2017-03-23T03:31:50.652Z",
    "build_snapshot" : false,
    "lucene_version" : "6.4.1"
  },
  "tagline" : "You Know, for Search"
}
* Illegal or missing hexadecimal sequence in chunked-encoding
* Curl_http_done: called premature == 1
* Closing connection 0
curl: (56) Illegal or missing hexadecimal sequence in chunked-encoding
```
So the mock Elasticsearch server seems to be trying to return the right thing, but seems to be doing so in the wrong way.

By contrast, doing the same thing against a real Elasticsearch cluster, I get the following (with no "content-encoding" or "transfer-encoding" headers):
```
$ curl -iv --raw http://es.test.yoyodyne.com:80/
*   Trying xx.xx.xx.xx...
* TCP_NODELAY set
* Connected to es.test.yoyodyne.com (xx.xx.xx.xx) port 80 (#0)
> GET / HTTP/1.1
> Host: es.test.yoyodyne.com
> User-Agent: curl/7.51.0
> Accept: */*
> 
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
Access-Control-Allow-Origin: *
< Content-Type: application/json; charset=UTF-8
Content-Type: application/json; charset=UTF-8
< Content-Length: 337
Content-Length: 337
< Connection: keep-alive
Connection: keep-alive

< 
{
  "name" : "7Sg9NBf",
  "cluster_name" : "041965370514:yoyod-test",
  "cluster_uuid" : "2ZY-wlZ3Txyr8BOHMm_2Cg",
  "version" : {
    "number" : "5.1.1",
    "build_hash" : "5395e21",
    "build_date" : "2016-12-15T22:47:19.049Z",
    "build_snapshot" : false,
    "lucene_version" : "6.3.0"
  },
  "tagline" : "You Know, for Search"
}
* Curl_http_done: called premature == 0
* Connection #0 to host es.test.yoyodyne.com left intact
```
So, **is there some configuration magic I should be providing, or something else I should do, to get the Elasticsearch server to either not use those headers, or to use them right?**

I should mention that accessing localstack's Elasticsearch via curl is not the final goal; ultimately I want to access it from a Go application that uses the "Elastic" client library (https://olivere.github.io/elastic/). That library reports "no Elasticsearch node available" when I try to connect to the container's Elasticsearch endpoint -- which probably isn't surprising if the same bad encoding is being returned.

**Comment by whummer** [comment]
Thanks for reporting. I believe this has to do with the default options that we use when spinning up Elasticsearch. See related issue here: https://github.com/atlassian/localstack/issues/102

We may have to add a configuration flag (e.g., `http.compression` and possibly others) to the command used to start Elasticsearch: https://github.com/atlassian/localstack/blob/master/localstack/mock/infra.py#L85-L87

Would you be able to look into this and prepare a pull request? Otherwise, I can try to take a look in the next couple of days. Thanks!

**Comment by whummer** [comment]
@AnvilStriker This should be fixed in commit 8124df3641ebc5854dae8c05911a6b3edc171c6e, Elasticsearch is now started with HTTP compression disabled.

Can you please pull the latest Docker image (tags `0.5.4` and `latest`) and give it a try? Please re-open this ticket if the problem persists. Thanks
